testng-core/src/main/java/org/testng/internal/Yaml.java

```diff
@@ -347,4 +352,42 @@ public Object construct(Node node) {
       }
     }
   }
+
+  private static class XmlClassTypeDescriptor extends TypeDescription {
+
+    private final boolean loadClasses;
+
+    public XmlClassTypeDescriptor(boolean loadClasses) {
+      super(XmlClass.class);
+      this.loadClasses = loadClasses;
+    }
+
+    @Override
+    public Object newInstance(Node node) {
+      String className;
+
+      try {
+        java.lang.reflect.Constructor<?> c =
+            XmlClass.class.getDeclaredConstructor(String.class, boolean.class);
+        c.setAccessible(true);
+        if (node instanceof MappingNode) {
+          Node valueNode =
+              ((MappingNode) node)
+                  .getValue().stream()
+                      .filter(
+                          nodeTuple ->
+                              ((ScalarNode) nodeTuple.getKeyNode()).getValue().equals("name"))
+                      .findFirst()
+                      .orElseThrow(RuntimeException::new)
```

Comment:
Can we consider throwing `org.testng.TestNGException` instead with a proper error message that explains what went wrong ?

Reply:
done